src/components/ConditionalFilter/UI/BulkSelect.tsx

```diff
@@ -0,0 +1,120 @@
+import { DynamicMultiselect } from "@saleor/macaw-ui-next";
+import React, { useEffect, useRef, useState } from "react";
+
+import { FilterEventEmitter } from "./EventEmitter";
+import { BulkselectOperator, RightOperatorOption } from "./types";
+
+export interface BulkSelectProps {
+  selected: BulkselectOperator;
+  emitter: FilterEventEmitter;
+  index: number;
+  error: boolean;
+  helperText: string;
+  disabled: boolean;
+}
+
+const truncate = ({ label, ...rest }: RightOperatorOption) => ({
+  label: label.length > 15 ? label.slice(0, 8) + "..." : label,
+  ...rest,
+});
+
+const sanitize = (text: string) => text.replace(/\s/g, ",");
+const clean = (text: string) => text !== "";
+const toOption = (text: string) => ({
+  label: text,
+  value: text,
+  slug: text,
+});
+
+const BulkSelect = ({ selected, emitter, index, error, helperText, disabled }: BulkSelectProps) => {
+  const [options, setOptions] = useState<RightOperatorOption[]>(selected.value ?? []);
+
+  const ref = useRef<HTMLInputElement>(null);
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (!ref.current) return;
+
+    const currentValue = (event.target as HTMLInputElement).value;
+
+    switch (event.key) {
+      case "Tab":
+      case "Enter":
+      case ",":
+      case " ":
+        event.preventDefault();
+        event.stopPropagation();
+
+        if (currentValue === "") {
+          return;
+        }
+
+        setOptions(prev => [...prev, toOption(currentValue)]);
+        ref.current.value = "";
+        ref.current.blur();
+        ref.current.focus();
+
+        return;
+      case "Backspace":
+        if (currentValue === "") {
+          setOptions(prev => prev.slice(0, prev.length - 1));
+        }
+
+        return;
+      default:
+        return;
+    }
+  };
+
+  useEffect(() => {
+    emitter.changeRightOperator(index, options);
```

Comment:
issue: events that are being emitted by the component don't have to be grouped into some sort of "emiter" field, components are meant to be UI-representation so it's fine when you pass props such as onFocusRIghtOperant etc.